### PR TITLE
Md zoom directive

### PR DESCRIPTION
**Pull request by cesar-garay**
Hi All, this is my first PR and I need som help.
I created a zoom directive for two images but I need a hand with the unit test.
Where can i found information to run gulp test task?
Thanks in advanced.

md-zoom-mov

Chrome
md-zoom-desk-chrome

FireFox
md-zoom-desk-firefox

IE 11
md-zoom-desk-ie11

### CODE REVIEW

File: src/components/zoom/zoom.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    $mdGesture.register(element, 'drag');
+    element
+      .on('$md.drag', onDrag)
+      .on('$md.dragstart', onDragStart)
+      .on('$md.dragend', onDragStop);
+
+
+    function onDragStop(ev) {
+      zoomedContainer.css('display', 'none  ');
```
Comment (mleanos) [inline src/components/zoom/zoom.js]: Remove extra spaces after "none"

File: src/components/zoom/zoom.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        this.$viewChangeListeners.forEach(function(cb) { cb(); });
+      },
+      $parsers: [],
+      $formatters: [],
+      $viewChangeListeners: []
+    };
+
+    var thumbContainer = angular.element(element[0].querySelector('.md-zoom-thumb-container'));
+    var zoomedContainer = angular.element(element[0].querySelector('.md-zoom-zoomed-container'));
+    var zoomeMessage = angular.element(element[0].querySelector('.md-zoom-message'));
```
Comment (mleanos) [inline src/components/zoom/zoom.js]: Should be `zoomedMessage`?

File: src/components/zoom/zoom.js (hunk 1 above)
Comment (cesar-garay) [inline src/components/zoom/zoom.js]: Done!

File: src/components/zoom/zoom.js (hunk 2 above)
Comment (cesar-garay) [inline src/components/zoom/zoom.js]: Typo error.. fixed

**Comment by EladBezalel** [comment]
@cesar-garay seems like a very nice feature!
but Angular-Material aims to develop an implementation for only the Material Design Spec,
Zoom is not part of it, but you should provide it as a separate module that other users can use.

Thanks for the effort and the hard work! we're really appreciate it!

**Comment by ThomasBurleson** [comment]
@cesar-garay - this seems like a nice extension to publish outside of Angular Material core.
> Note: please do not use the `md-` namespace prefix for components. That namespace is for Angular team's core use and helps us manage versioning and conflicts. You could, however, easily use `mdx-`.

**Comment by cesar-garay** [comment]
Thanks @EladBezalel , @ThomasBurleson I'll create a separate extension.
Thanks so much for your answer!!

**Comment by cesar-garay** [comment]
Hi @EladBezalel, @ThomasBurleson .
I create a separated library for this feature here https://cesar-garay.github.io/zoom-directive/build/.
Could you take a look on it?
Thanks!

**Pull request closed without merging**